src/renault_api/kamereon/models.py

```diff
@@ -110,57 +121,53 @@ class KamereonVehiclesDetailsGroup(BaseModel):
     brand: Optional[KamereonVehiclesDetailsGroup]
     model: Optional[KamereonVehiclesDetailsGroup]
     energy: Optional[KamereonVehiclesDetailsGroup]
 
-    def get_vin(self) -> str:
-        """Return vehicle vin."""
-        if self.vin is None:  # pragma: no cover
-            raise KamereonException("`vin` is None in KamereonVehiclesDetails.")
-        return self.vin
-
-    def get_registration_number(self) -> str:
-        """Return vehicle vin."""
-        if self.registrationNumber is None:  # pragma: no cover
-            raise KamereonException(
-                "`registrationNumber` is None in KamereonVehiclesDetails."
-            )
-        return self.registrationNumber
-
-    def get_energy_code(self) -> EnergyCode:
+    def get_energy_code(self) -> Optional[enums.EnergyCode]:
         """Return vehicle energy code."""
         if self.energy is None:  # pragma: no cover
-            raise KamereonException("`energy` is None in KamereonVehiclesDetails.")
+            return None
         if self.energy.code is None:  # pragma: no cover
-            raise KamereonException("`energy.code` is None in KamereonVehiclesDetails.")
+            return None
         try:
-            return EnergyCode(self.energy.code)
+            return enums.EnergyCode(self.energy.code)
         except ValueError:  # pragma: no cover
-            raise KamereonException(
+            raise exceptions.KamereonException(
                 f"Unable to convert `{self.energy.code}` to EnergyCode."
             )
 
-    def get_brand_label(self) -> str:
+    def get_brand_label(self) -> Optional[str]:
         """Return vehicle model label."""
         if self.brand is None:  # pragma: no cover
-            raise KamereonException("`brand` is None in KamereonVehiclesDetails.")
-        if self.brand.label is None:  # pragma: no cover
-            raise KamereonException("`brand.label` is None in KamereonVehiclesDetails.")
+            return None
         return self.brand.label
 
-    def get_model_label(self) -> str:
+    def get_model_code(self) -> Optional[str]:
+        """Return vehicle model code."""
+        if self.model is None:  # pragma: no cover
+            return None
+        return self.model.code
+
+    def get_model_label(self) -> Optional[str]:
         """Return vehicle model label."""
         if self.model is None:  # pragma: no cover
-            raise KamereonException("`model` is None in KamereonVehiclesDetails.")
-        if self.model.label is None:  # pragma: no cover
-            raise KamereonException("`model.label` is None in KamereonVehiclesDetails.")
+            return None
         return self.model.label
 
+    def reports_charging_power_in_watts(self) -> bool:
+        """Return True if model reports chargingInstantaneousPower in watts."""
+        if self.get_model_code() in [
+            "X101VE",  # ZOE 40
+        ]:
+            return True
+        return False
```

Comment:
No way to get it from the API ?

Reply:
Until we have some more samples, then this is the only way that I know of. There was a discussion on PyZE about it...
It seems that Zoe50 reports in kW, and I expect all newer models will also default to kW.